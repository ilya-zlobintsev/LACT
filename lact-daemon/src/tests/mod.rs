use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
