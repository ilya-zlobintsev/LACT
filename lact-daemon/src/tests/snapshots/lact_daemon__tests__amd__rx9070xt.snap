---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "max_mclk": 1259,
    "table": {
      "type": "amd",
      "value": {
        "data": {
          "current_mclk_range": {
            "max": 1259,
            "min": 97
          },
          "current_sclk_range": {
            "max": null,
            "min": null
          },
          "od_range": {
            "curve_sclk_points": [],
            "curve_voltage_points": [],
            "mclk": {
              "max": 1500,
              "min": 97
            },
            "sclk": null,
            "sclk_offset": {
              "max": 1000,
              "min": -500
            },
            "voltage_offset": {
              "max": 0,
              "min": -200
            }
          },
          "sclk_offset": 0,
          "vddc_curve": [],
          "voltage_offset": -50
        },
        "kind": "rdna"
      }
    }
  },
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "32.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 48 [RX 9070/9070 XT]",
        "model_id": "7550",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "PowerColor Hellhound/Reaper/Red Devil RX 9070 XT",
        "model_id": "2435",
        "vendor": "Tul Corporation / PowerColor",
        "vendor_id": "148C"
      }
    },
    "vbios_version": "113-EXT109069-101",
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 48 [RX 9070/9070 XT]",
      "model_id": "7550",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "PowerColor Hellhound/Reaper/Red Devil RX 9070 XT",
      "model_id": "2435",
      "vendor": "Tul Corporation / PowerColor",
      "vendor_id": "148C"
    }
  },
  "power_profile_modes": {
    "active": 0,
    "modes": {
      "0": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "BOOTUP_DEFAULT"
      },
      "1": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5570560,
              -655,
              -6553
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "3D_FULL_SCREEN"
      },
      "2": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              3,
              0,
              5898240,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5898240,
              -65536,
              -6553
            ]
          }
        ],
        "name": "POWER_SAVING"
      },
      "3": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              4,
              500,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "VIDEO"
      },
      "4": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              4,
              600,
              1,
              0,
              4587520,
              -16384,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "VR"
      },
      "5": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              4,
              600,
              1,
              0,
              4587520,
              -16384,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "COMPUTE"
      },
      "6": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "CUSTOM"
      },
      "7": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              1,
              0,
              1,
              0,
              5570560,
              -655,
              -6553
            ]
          },
          {
            "clock_type": "FCLK",
            "values": [
              0,
              3,
              0,
              1,
              0,
              5898240,
              -6553,
              -6553
            ]
          }
        ],
        "name": "WINDOW_3D"
      }
    },
    "value_names": [
      "FPS",
      "MinActiveFreqType",
      "MinActiveFreq",
      "BoosterFreqType",
      "BoosterFreq",
      "PD_Data_limit_c",
      "PD_Data_error_coeff",
      "PD_Data_error_rate_coeff"
    ]
  },
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 500
      },
      {
        "enabled": true,
        "index": 1,
        "value": 16
      },
      {
        "enabled": true,
        "index": 2,
        "value": 2400
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 96
      },
      {
        "enabled": true,
        "index": 1,
        "value": 456
      },
      {
        "enabled": true,
        "index": 2,
        "value": 772
      },
      {
        "enabled": true,
        "index": 3,
        "value": 875
      },
      {
        "enabled": true,
        "index": 4,
        "value": 1124
      },
      {
        "enabled": true,
        "index": 5,
        "value": 1258
      }
    ]
  },
  "stats": {
    "busy_percent": 3,
    "clockspeed": {
      "gpu_clockspeed": 167,
      "vram_clockspeed": 1258
    },
    "core_power_state": 1,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {
        "acoustic_limit": {
          "allowed_range": [
            500,
            3650
          ],
          "current": 3300
        },
        "acoustic_target": {
          "allowed_range": [
            500,
            3650
          ],
          "current": 1000
        },
        "minimum_pwm": {
          "allowed_range": [
            30,
            100
          ],
          "current": 30
        },
        "target_temp": {
          "allowed_range": [
            25,
            105
          ],
          "current": 92
        },
        "zero_rpm_enable": false,
        "zero_rpm_temperature": {
          "allowed_range": [
            50,
            110
          ],
          "current": 67
        }
      },
      "pwm_current": 81,
      "pwm_max": 254,
      "pwm_min": 76,
      "speed_current": 896,
      "speed_max": 3650,
      "speed_min": 0,
      "temperature_range": [
        25,
        100
      ]
    },
    "memory_power_state": 5,
    "pcie_power_state": 1,
    "performance_level": "auto",
    "power": {
      "average": 55.0,
      "cap_current": 120.0,
      "cap_default": 304.0,
      "cap_max": 340.0,
      "cap_min": 0.0
    },
    "temps": {
      "edge": {
        "crit": 110.0,
        "crit_hyst": -273.1499938964844,
        "current": 42.0
      },
      "junction": {
        "crit": 110.0,
        "crit_hyst": -273.1499938964844,
        "current": 45.0
      },
      "mem": {
        "crit": 108.0,
        "crit_hyst": -273.1499938964844,
        "current": 68.0
      }
    },
    "voltage": {
      "gpu": 552
    },
    "vram": {}
  }
}
